nncf/quantization/algorithms/weight_compression/torch_fx_backend.py

```diff
@@ -0,0 +1,239 @@
+            channel_idx = (
+                1
+                if node_with_weight.metatype
+                in [om.PTConvTranspose1dMetatype, om.PTConvTranspose2dMetatype, om.PTConvTranspose3dMetatype]
+                else 0
+            )
+            reduction_axes = [i for i in range(ndims) if i != channel_idx]
+        return tuple(reduction_axes)
+
+    @staticmethod
+    def target_point(target_type: TargetType, target_node_name: str, port_id: int) -> PTTargetPoint:
+        return PTWeightCompressionAlgoBackend.target_point(target_type, target_node_name, port_id)
+
+    @staticmethod
+    def raw_statistic_collector(num_samples: Optional[int] = None) -> TensorCollector:
+        return get_raw_stat_collector(num_samples)
+
+    @staticmethod
+    def get_activation_port_id(node: NNCFNode, graph: NNCFGraph) -> int:
+        return PTWeightCompressionAlgoBackend.get_activation_port_id(node, graph)
+
+    def get_weight(
+        self, node_with_weight: NNCFNode, weight_port_id: int, model: torch.fx.GraphModule, graph: NNCFGraph
+    ) -> Tensor:
+        weight_edge = graph.get_input_edge_by_port_id(node_with_weight, weight_port_id)
+        weight_node = weight_edge.from_node
+        graph_weight_node = get_graph_node_by_name(model.graph, weight_node.node_name)
+        weight = get_tensor_constant_from_node(graph_weight_node, model).data
+        if weight is None:
+            raise nncf.InternalError(f"Could not find a node in the model by name {weight_node}.")
+
+        return Tensor(weight)
+
+    def get_weight_dtype(
+        self, node_with_weight: NNCFNode, weight_port_id: int, model: torch.fx.GraphModule, graph: NNCFGraph
+    ) -> TensorDataType:
+        return self.get_weight(node_with_weight, weight_port_id, model, graph).dtype
+
+    @staticmethod
+    def get_weight_shape(node_with_weight: NNCFNode, weight_port_id: int, graph: NNCFGraph) -> Tuple:
+        weight_node = get_const_node(node_with_weight, weight_port_id, graph)
+        edge = graph.get_edge(weight_node, node_with_weight)
+        return tuple(edge.tensor_shape)
+
+    def set_weight(
+        self,
+        node_with_weight: NNCFNode,
+        weight_port_id: int,
+        model: torch.fx.GraphModule,
+        graph: NNCFGraph,
+        weight: Tensor,
+    ) -> None:
+        constant_update_transformation_builder(node_with_weight, weight.data, input_port_id=weight_port_id)(model)
+
+    def insert_adapters(
+        self, wc_params: WeightCompressionParameters, lora_A: Tensor, lora_B: Tensor, int8_lora: bool
+    ) -> None:
+        pass
+
+    def transform_model(
+        self,
+        model: torch.fx.GraphModule,
+        graph: NNCFGraph,
+        weight_compression_parameters: Iterable[WeightCompressionParameters],
+        precomputed_scales: Dict[str, Tensor] = None,
+        precomputed_zero_points: Dict[str, Tensor] = None,
+        lora_correction_algo: LoraCorrectionAlgorithm = None,
+    ) -> torch.fx.GraphModule:
```

Comment:
Perhaps we can put common logic of PT/FX backends in a function and reuse it in both backends

Reply:
Yes, that would be nice I think. Would you recommend a helpers.py or something of that sort? or maybe a private method in one of the backends with the reusable code?

Reply:
Replying to [this](https://github.com/openvinotoolkit/nncf/pull/2891#discussion_r1767285048), but that could maybe disrupt the structure of the backends while possibly making the code less streamlined and maintainable in the future right?